### ISSUE TITLE
notification log

### ISSUE THREAD

**Issue by annamorphism** [DDMAL/Neon#950]
I like that the error messages are now grouped and possible to review!

But:

1) The discrepancy of name bugs me--"errors" and "notifications" are not the same, and the casual user might think we're going to highlight problems with the neumes or something (ha, ha! that's *their* job!)

2) should this maybe be under MEI Actions rather than as a "view"? I feel like it's something that's more for testers than "normal" users (since ideally the MEI doesn't have any errors in the first place.)

**Comment by jinh0** [comment]
Hey @annamorphism the notification log has now been renamed to error log. Although I am not sure what the consensus is about whether to put this under MEI actions. Technically this isn't an "MEI action".

Maybe we should have an option to show "advanced settings"? Which when enabled, it shows the status of the MEI (which I remember @fujinaga wanted to not be for anyone outside of testers and developers), the error log option, MEI actions, and more.

**Comment by fujinaga** [comment]
I like the "advanced settings' idea. Make sure to warn the user that this is only for developers.

**Comment by annamorphism** [comment]
Yes to an "advanced setting" mode! Or maybe "developer settings" or something like that. (Also means we avoid people looking for problems with their neumes under "error log".)  MEI status/actions/notifications all sort of go together, but only for select people.

**Comment by jinh0** [comment]
Just to confirm, what should go under advanced settings? Currently I have in mind,

- MEI validation
- MEI actions
- Error log

**Comment by annamorphism** [comment]
@jinh0 would making all MEI actions "advanced" also get rid of "Revert" for most users? Wondering if that's desirable. 
Otherwise I think that's the right bunch of things to include. 
Are you imagining that in "Advanced settings" the error log always appears (like the MEI indicator) and otherwise it never does? Or that the current "Errors" display check box only appears in Advanced settings (and otherwise doesn't)--so you have the option of turning it on and off? Doesn't really matter to me which it is, but if it's the former, the checkbox will end up redundant.  
Also, is this something you turn on *after* loading the page? If so, when do the notifications themselves float into view: do they show up for everyone upon page load (but you can only see the log/use actions on them after turning on Advanced Settings)? Or do you turn on Advanced Settings, and *then* the checks run and give you all the notifications/errors?  (I think that makes more sense, personally.)

**Comment by jinh0** [comment]
> Or do you turn on Advanced Settings, and then the checks run and give you all the notifications/errors? (I think that makes more sense, personally.)

I think that makes sense, that the checks related to the MEI actions should run AFTER you turn on advanced settings.

> the current "Errors" display check box only appears in Advanced settings (and otherwise doesn't)--so you have the option of turning it on and off?

Yeah, this was what I was thinking. If you have "Advanced settings" enabled, then the error display checkbox is shown. So you still have the choice to not turn them on. From what I understand, "advanced settings" is like turning off a filter, and you get more options, but it's up to you whether to use them.

**Comment by JoyfulGen** [comment]
We have a Debug mode that allows the user to see all error logs! Closing issue.
